Null pointer passed to 1st parameter expecting 'nonnull' [clang-analyzer-core.NonNullParamChecker]

```
/home/runner/work/coturn/coturn/src/apps/relay/dbdrivers/dbd_sqlite.c:167:18: warning: Null pointer passed to 1st parameter expecting 'nonnull' [clang-analyzer-core.NonNullParamChecker]
    size_t szh = strlen(home);
                 ^
/home/runner/work/coturn/coturn/src/apps/relay/dbdrivers/dbd_sqlite.c:1281:31: note: Calling 'get_sqlite_connection'
  sqlite3 *sqliteconnection = get_sqlite_connection();
                              ^~~~~~~~~~~~~~~~~~~~~~~
/home/runner/work/coturn/coturn/src/apps/relay/dbdrivers/dbd_sqlite.c:214:7: note: Assuming 'sqliteconnection' is null
  if (!sqliteconnection) {
      ^~~~~~~~~~~~~~~~~
/home/runner/work/coturn/coturn/src/apps/relay/dbdrivers/dbd_sqlite.c:214:3: note: Taking true branch
  if (!sqliteconnection) {
  ^
/home/runner/work/coturn/coturn/src/apps/relay/dbdrivers/dbd_sqlite.c:216:5: note: Calling 'fix_user_directory'
    fix_user_directory(pud->userdb);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/runner/work/coturn/coturn/src/apps/relay/dbdrivers/dbd_sqlite.c:150:10: note: Assuming the condition is false
  while (*dir == ' ')
         ^~~~~~~~~~~
/home/runner/work/coturn/coturn/src/apps/relay/dbdrivers/dbd_sqlite.c:150:3: note: Loop condition is false. Execution continues on line 153
  while (*dir == ' ')
  ^
/home/runner/work/coturn/coturn/src/apps/relay/dbdrivers/dbd_sqlite.c:153:7: note: Assuming the condition is true
  if (*dir == '~') {
      ^~~~~~~~~~~
/home/runner/work/coturn/coturn/src/apps/relay/dbdrivers/dbd_sqlite.c:153:3: note: Taking true branch
  if (*dir == '~') {
  ^
/home/runner/work/coturn/coturn/src/apps/relay/dbdrivers/dbd_sqlite.c:154:18: note: Assuming pointer value is null
    char *home = getenv("HOME");
                 ^~~~~~~~~~~~~~
/home/runner/work/coturn/coturn/src/apps/relay/dbdrivers/dbd_sqlite.c:154:5: note: 'home' initialized here
    char *home = getenv("HOME");
    ^~~~~~~~~~
/home/runner/work/coturn/coturn/src/apps/relay/dbdrivers/dbd_sqlite.c:155:10: note: 'home' is null
    if (!home) {
         ^~~~
/home/runner/work/coturn/coturn/src/apps/relay/dbdrivers/dbd_sqlite.c:155:5: note: Taking true branch
    if (!home) {
    ^
/home/runner/work/coturn/coturn/src/apps/relay/dbdrivers/dbd_sqlite.c:157:11: note: Assuming 'pwd' is null
      if (!pwd) {
          ^~~~
/home/runner/work/coturn/coturn/src/apps/relay/dbdrivers/dbd_sqlite.c:157:7: note: Taking true branch
      if (!pwd) {
      ^
/home/runner/work/coturn/coturn/src/apps/relay/dbdrivers/dbd_sqlite.c:167:18: note: Null pointer passed to 1st parameter expecting 'nonnull'
    size_t szh = strlen(home);
                 ^      ~~~~
```

--- a/src/apps/relay/dbdrivers/dbd_sqlite.c
+++ b/src/apps/relay/dbdrivers/dbd_sqlite.c
@@ -156,6 +156,7 @@ static void fix_user_directory(char *dir0) {
       struct passwd *pwd = getpwuid(getuid());
       if (!pwd) {
         TURN_LOG_FUNC(TURN_LOG_LEVEL_ERROR, "Cannot figure out the user's HOME directory (1)\n");
+        return;
       } else {
         home = pwd->pw_dir;
         if (!home) {

